TINY-9517: Sidebar element accessibility fixes (#8451)
tinymce/tinymce: modules/tinymce/CHANGELOG.md

- The `link` plugins context menu items will no longer appear for noneditable links. #TINY-9491
- The formatting of `contenteditable="false"` elements are no longer cloned to new cells while creating new table rows. #TINY-9449
- Changed the color of `@dialog-table-border-color`, and added right padding to the first cell of dialog table. #TINY-9380

> - The sidebar element now have attribute `role="region"` when visible and attribute `role="presentation"` when hidden. #TINY-9517

Reviewer: ```suggestion
- The sidebar element now has accessibility role `region` when visible and `presentation` when hidden. #TINY-9517
```
